OpenRA.Mods.Common/Traits/AffectsShroud.cs

```diff
@@ -64,12 +65,14 @@ public void Tick(Actor self)
 			var projectedPos = centerPosition - new WVec(0, centerPosition.Z, centerPosition.Z);
 			var projectedLocation = self.World.Map.CellContaining(projectedPos);
 			var disabled = IsDisabled(self);
+			var traitDisabled = IsTraitDisabled;
 
-			if (cachedLocation == projectedLocation && cachedDisabled == disabled)
+			if (cachedLocation == projectedLocation && traitDisabled == cachedTraitDisabled && cachedDisabled == disabled)
 				return;
 
 			cachedLocation = projectedLocation;
 			cachedDisabled = disabled;
+			cachedTraitDisabled = traitDisabled;
 
 			var cells = ProjectedCells(self);
 			foreach (var p in self.World.Players)
```

Comment:
It's checked in line 70, otherwise it just serves to update `cachedTraitDisabled` in line 75.